unpacker.go

```diff
@@ -137,8 +137,15 @@ EachLayer:
 		)
 
 		for try := 1; try <= 3; try++ {
+			// If we want to avoid unpacking the same layer in parallel, we need to use a key format that will
+			// intentionally cause collisions for the same layer. This is how we'll know an active snapshot is
+			// underway already, along with intentional tracking in the metadata snapshotter layer.
+			if rCtx.DisableSameLayerUnpack {
```

Comment:
Looks like this `UnpackKeyFormat` is used at one more place too (https://github.com/containerd/containerd/blob/main/rootfs/apply.go#L125), should we update that location too?

Reply:
This was kind of a "fix it for cri" approach and cri ends up in unpacker.go here

Reply:
Hmm I agree but I think we should at least try to make sure that the snapshot key uses same formatting everywhere. Otherwise we might end up having same image under different keys?

Reply:
Sure, let me reason with if anything else needs to change besides just the key format in that code then. Does ctr hit that code path I'd imagine?

Reply:
Yeah, IIRC ctr import goes through that.

Reply:
So re-thinking on this, if ctr (or anything else) hits this code path I don't think it's that troublesome. From an actual production standpoint we're going to go through cri so we'll hit unpack.go mainly, but even in the case where someone got on a machine and ran ctr locally to troubleshoot something it should prepare  the layers with that unique key first, and then when going to actually commit it it should get an error as theres already a committed snapshot with that hash. 

I'm not as familiar with that code path but if before it even gets to applyLayers it stats for a snapshot with a hash of the layer it should bail out also even earlier, as it should exist. Now if you kicked off a ctr import/pull and a cri pull at the same time, yes there'd be a bit of a race to the finish as there is now and the ctr ones wouldn't be waiting for the other to complete, but I don't think that's a big worry really.

Reply:
PR feedback commit just has a small change Maksim had brought up to gate the extra abort() calls behind the DisableSameLayerUnpack field, let me know if the above is still a concern ^